ads/google/a4a/test/test-google-ads-a4a-config.js

```diff
@@ -310,6 +312,261 @@ describe('a4a_config', () => {
+      let externalBranches, internalBranches;
+      if (isExperimentOn(win, 'a4aFastFetchDoubleclickLaunched')) {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+      } else {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+      }
+      expect(googleAdsIsA4AEnabled(win, element, 'expDoubleclickA4A',
+        externalBranches, internalBranches)).to.be.true;
+    });
+  it('should serve Delayed Fetch to launched DoubleClick experiment (holdback)',
+    () => {
+      toggleExperiment(win, 'a4aFastFetchDoubleclickLaunched', true, true);
+      toggleExperiment(win, 'expDoubleclickA4A', true, true);
+      element.setAttribute('type', 'doubleclick');
+      const postLaunch =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+      win.pageExperimentBranches['expDoubleclickA4A'] = postLaunch.experiment;
+      let externalBranches, internalBranches;
+      if (isExperimentOn(win, 'a4aFastFetchDoubleclickLaunched')) {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+      } else {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+      }
+      expect(googleAdsIsA4AEnabled(win, element, 'expDoubleclickA4A',
+        externalBranches, internalBranches)).to.be.false;
+    });
+  it('should serve Delayed Fetch to unlaunched DoubleClick filler (URL)',
+    () => {
+      win.location.search = '?exp=a4a:0';
+      toggleExperiment(win, 'a4aFastFetchDoubleclickLaunched', false, true);
+      element.setAttribute('type', 'doubleclick');
+      let externalBranches, internalBranches;
+      if (isExperimentOn(win, 'a4aFastFetchDoubleclickLaunched')) {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH;
+      } else {
+        externalBranches =
+          expts.DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+        internalBranches =
+          expts.DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH;
+      }
+      expect(googleAdsIsA4AEnabled(win, element, EXP_ID,
+        externalBranches, internalBranches)).to.be.false;
+      expect(win.document.cookie).to.be.null;
+    });
+  it('should serve Delayed Fetch to unlaunched DoubleClick control (URL)',
+    () => {
+      win.location.search = '?exp=a4a:1';
```

Comment:
I'm a little confused by this, as googleAdsIsA4AEnabled is calling maybeSetExperimentsFromUrl already. Let's discuss in the morning. Also note that I moved these methods back to the other file.